### PR TITLE
[Surrey] Add External ID to CSV export

### PR DESCRIPTION
**Pull request by chrismytton**
For FD-4770

### CODE REVIEW

File: perllib/FixMyStreet/Cobrand/Surrey.pm [hunk 1]
```diff
@@ -200,18 +200,18 @@ sub dashboard_export_problems_add_columns {
...
         my $report = shift;
 
-        if ($alerts_lookup) {
-            return { alerts_count => ($alerts_lookup->{$report->id} || 0) };
-        } else {
-            return { alerts_count => ($report->{alerts_count} || 0) };
-        }
+        return {
+            alerts_count => ($alerts_lookup ? $alerts_lookup->{$report->id} || 0 : $report->{alerts_count} || 0),
+            external_id  => $csv->dbi ? $report->{external_id} : $report->external_id,
```
Comment (chrismytton) [inline perllib/FixMyStreet/Cobrand/Surrey.pm:213]: Is this right? I notice that Northamptonshire just does `return if $csv->dbi;`, but couldn't work our why 😕

Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Surrey.pm:213]: The Script::CSVExport gets all the columns on a report by default in its hashref, but the non-pre-gen only gets a subset. So you don't need to look up anything that's a column of the DB when $csv->dbi is set, you could if $alerts_lookup is set add external_id there, and if not set return only alerts_count as it does currently.

**Review by dracos — CHANGES_REQUESTED** [review]
Works fine, but yes, from your comment I think you can leave it more like it was before, only adding a line to csv_extra_data in the alerts_lookup being set section.

**Review by dracos — APPROVED** [review]
Thanks :)

**Pull request merged**